src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldHaveCorrectSuffix.cs

```diff
@@ -25,24 +29,141 @@ public abstract class IdentifiersShouldHaveCorrectSuffixAnalyzer : DiagnosticAna
+                                                        Tuple.Create("System.EventArgs", false, "EventArgs", false),
+                                                        Tuple.Create("System.Exception", false, "Exception", false),
+                                                        Tuple.Create("System.Collections.ICollection", true, "Collection", false),
+                                                        Tuple.Create("System.Collections.IDictionary", true, "Dictionary", false),
+                                                        Tuple.Create("System.Collections.IEnumerable", true, "Collection", false),
+                                                        Tuple.Create("System.Collections.Queue", false, "Queue", true),
+                                                        Tuple.Create("System.Collections.Stack", false, "Stack", true),
+                                                        Tuple.Create("System.Collections.Generic.Queue`1", false, "Queue", true),
+                                                        Tuple.Create("System.Collections.Generic.Stack`1", false, "Stack", true),
+                                                        Tuple.Create("System.Collections.Generic.ICollection`1", true,"Collection", false),
+                                                        Tuple.Create("System.Collections.Generic.IDictionary`2", true, "Dictionary", false),
+                                                        Tuple.Create("System.Data.DataSet", false, "DataSet", false),
+                                                        Tuple.Create("System.Data.DataTable", false, "DataTable", true),
+                                                        Tuple.Create("System.IO.Stream", false, "Stream", false),
+                                                        Tuple.Create("System.Security.IPermission", true,"Permission", false),
+                                                        Tuple.Create("System.Security.Policy.IMembershipCondition", true, "Condition", false)
+                                                    };
+
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(AnalyzeCompilationStart);
+        }
+
+        private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var baseTypeSuffixMap = new Dictionary<INamedTypeSymbol, SuffixInfo>();
+            var interfaceTypeSuffixMap = new Dictionary<INamedTypeSymbol, SuffixInfo>();
+
+            foreach (var tuple in s_baseTypesAndTheirSuffix)
+            {
+                var wellKnownNamedType = context.Compilation.GetTypeByMetadataName(tuple.Item1);
+
+                if (wellKnownNamedType != null && wellKnownNamedType.OriginalDefinition != null)
+                {
+                    // If the type is interface
+                    if (tuple.Item2)
```

Comment:
I can get the value of `tuple.Item2` from `wellKnownNamedType`. Will modify that